src/server/rest/v1/service/TagService.ts

```diff
@@ -143,6 +143,18 @@ export default class TagService {
         action: action
       });
     }
+    // Check Tag
+    tag = await TagStorage.getTagByVisualID(req.user.tenantID, filteredRequest.visualID);
+    if (tag) {
+      throw new AppError({
+        source: Constants.CENTRAL_SERVER,
+        errorCode: HTTPError.TAG_VISUAL_ID_ALREADY_EXIST_ERROR,
+        message: `Tag with ID '${filteredRequest.id}' already exists`,
```

Comment:
The error message is misleading: you look for for duplicate visualID records and the message is refereeing to the tag id.